addons/account/tests/test_account_move_send.py

```diff
@@ -708,6 +710,38 @@ def test_invoice_multi_email_missing(self):
         self.assertTrue(invoice2.invoice_pdf_report_id)
         self.assertTrue(self._get_mail_message(invoice2))
 
+    def test_invoice_multi_with_edi(self):
+        invoice1 = self.init_invoice("out_invoice", partner=self.partner_a, amounts=[1000], post=True)
+        invoice2 = self.init_invoice("out_invoice", partner=self.partner_b, amounts=[1000], post=True)
+
+        self.partner_a.invoice_sending_method = 'email'
+        self.partner_b.invoice_sending_method = 'manual'
+
+        def get_default_extra_edis(self, move):
+            if move == invoice1:
+                return {'edi1'}
+            else:
+                return {'edi1', 'edi2'}
+
+        def get_all_extra_edis(self):
+            return {
+                'edi1': {'label': 'EDI 1'},
+                'edi2': {'label': 'EDI 2'},
+            }
+
+        with (
+            patch.object(AccountMoveSend, '_get_default_extra_edis', get_default_extra_edis),
+            patch.object(AccountMoveSend, '_get_all_extra_edis', get_all_extra_edis)
```

Comment:
You can also just patch the class, which doesn't require any import. Is there any reason you choose to patch.object ? :thinking: 
`with patch('odoo.addons.account.models.account_move_send.AccountMoveSend.any_method', _any_method_remplacement):`